x/emissions/module/rewards/rewards.go

```diff
@@ -168,13 +168,13 @@ func GenerateRewardsDistributionByTopic(
 				if err != nil {
 					return nil, errors.Wrapf(err, "failed to add weight to sum")
 				}
-			}
-
-			// This way we won't double count from this earlier epoch revenue the next epoch
-			// This must come after GetTopicFeeRevenue() is last called per topic because otherwise the returned revenue will be zero
-			err = k.ResetTopicFeeRevenue(ctx, topicId, blockHeight)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to reset topic fee revenue")
+			} else {
+				// This way we won't double count from this earlier epoch revenue the next epoch
```

Comment:
well, we're clearly doing the opposite of these comments now

Reply:
right but the "double counting" will be for topics that haven't been paid out rewards yet!

Reply:
do you want me to just change the comment or do you want to do something else / handle this somehow else

Reply:
just change the comment cuz we are "double counting" in the sense of the original comment. like, just explain why we're doing this control flow